### PR TITLE
[MAINTENANCE] Remove Prefix "Batch" from "BatchSorter" and "BatchSortersDefinition" (in New Datasources)

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
Prefix "Batch" in "BatchSorter" and "BatchSortersDefinition" is not needed -- it is clear from context (and modules are name-spaced).

### Remark
This pull request can be reviewed after #7281 is merged (in order for the number of files to be reviewed to be the smallest).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: /GREAT-1462/GREAT-1737
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/experimental/datasources/interfaces.py [hunk 1]
```diff
@@ -296,15 +293,22 @@ def _compare_function(a: Batch, b: Batch) -> int:
...
-        elif a.metadata[key] is None:  # b.metadata[key] is not None
+
+        if a.metadata[key] is None:  # b.metadata[key] is not None
             return -1
-        # b.metadata[key] is None, a.metadata[key] is not None
+
+        if a.metadata[key] is not None:  # b.metadata[key] is None
+            return 1
+
+        # This line should never be reached.
```
Comment (billdirks) [inline great_expectations/experimental/datasources/interfaces.py]: Can we raise here so we catch it if this line is reached?

File: tests/experimental/datasources/test_config.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 )
 
 try:
-    from devtools import debug as pp
+    from devtools import debug as pp  # noqa: disable=E0602
```
Comment (billdirks) [inline tests/experimental/datasources/test_config.py]: I wonder why this error disable is necessary now when it wasn't before. Could you declare `pp` before this try block to remove this `noqa`? That is `pp: Callable[...]` and set `...` to whatever it should be.

**Review by billdirks — CHANGES_REQUESTED** [review]
I have 2 small comments but otherwise LGTM. Thanks for renaming!

File: tests/experimental/datasources/test_config.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/experimental/datasources/test_config.py]: @billdirks The `noqa` is needed because to disable `PyCharm` warning (which says "devtools in try block should also be in the except block"); however, I also declared the `Callable` based on your suggestion.  Thanks.

File: great_expectations/experimental/datasources/interfaces.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/experimental/datasources/interfaces.py]: @billdirks Good idea -- done.  Thanks.

File: tests/experimental/datasources/test_config.py (hunk 2 above)
Comment (Kilo59) [reply tests/experimental/datasources/test_config.py]: It shouldn't be needed. This is a PyCharm warning, not a CI linting or type-checker warning.
@alexsherstinsky I think you should revert this change.

As a sidenote this try import shouldn't be needed at all anymore. It was useful for some early debugging.

File: tests/experimental/datasources/test_config.py [hunk 3]
```diff
@@ -229,7 +232,7 @@ def test_from_gx_config(self, asset_dict: dict):
         gx_config = GxConfig.parse_obj({"xdatasources": {"my_ds": ds_dict}})
 
         gx_config_dict = gx_config.dict()
-        print(f"gx_config_dict\n{pf(gx_config_dict)}")
+        print(f"gx_config_dict\n{pprint.pformat(gx_config_dict)}")
```
Comment (Kilo59) [inline tests/experimental/datasources/test_config.py]: Please revert these changes to the `pprint` usage.

You can remove the `try import` of `devtools import debug as pp` to just a normal `from pprint import pprint as pp` if that makes your IDE happy.

Comment (alexsherstinsky) [reply tests/experimental/datasources/test_config.py]: @Kilo59 Just curious, how come the fully-quailfied way is not good?  Thanks.

File: tests/experimental/datasources/test_config.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/experimental/datasources/test_config.py]: @Kilo59 Commited -- thanks for pairing and the devtools docs!

File: great_expectations/experimental/datasources/interfaces.py [hunk 4]
```diff
@@ -296,16 +293,25 @@ def _compare_function(a: Batch, b: Batch) -> int:
...
-        # b.metadata[key] is None, a.metadata[key] is not None
-        return 0
+
+        if a.metadata[key] is not None:  # b.metadata[key] is None
+            return 1
+
+        # This line should never be reached; hence, "ValueError" with corresponding error message is raised.
+        raise ValueError(
+            f'Unexpected Batch metadata key combination, "{a.metadata[key]}" and "{b.metadata[key]}", was encountered.'
+        )
```
Comment (Kilo59) [inline great_expectations/experimental/datasources/interfaces.py:314]: I would call out that this change as being a runtime change above and beyond the rest of this PR which is just about renaming things.

Comment (alexsherstinsky) [reply great_expectations/experimental/datasources/interfaces.py:314]: @Kilo59 Yes, in a conversation with @billdirks we agreed that I can go ahead and fix this (and I just put it here as I was right there in the code), but yes -- it is in addition to just the name change.  Thanks.

**Review by billdirks — APPROVED** [review]
LGTM.
